src/lib.rs

```diff
@@ -98,7 +98,16 @@ impl<'a> TableOfContents<'a> {
     /// let toc = TableOfContents::new("# Heading\n");
     /// ```
     pub fn new(text: &'a str) -> Self {
-        let events = Parser::new_ext(text, CmarkOptions::all());
+        // We are not enabling all options since we want to mimic
+        // GitHub's behavior as closely as possible by default.
+        // And e.g. enabling smart punctuation would result in
+        // ... becoming … in the heading anchor (which GitHub doesn't do)
```

Comment:
Could you add a test for this case?

Reply:
Sure ... if I do this now however this PR will conflict with #6 ... I'd rather have the other PR be merged first so that I can rebase this PR and avoid the conflict.

Reply:
Or I guess I could just rebase it here and then later drop the other commits when they have been merged ... sigh ... why doesn't GitHub support stacked PRs is beyond me.

Reply:
GitHub sort of does support it, if you change the base of this PR to be the other branch then when you merge the first one it will automatically change the base of this one to be the default branch.

Reply:
The base branch of the PR has to be in this repository, so no I can't do that since I don't have push access here.

Reply:
Ok, I was actually mistaken about smart punctuation being able to result in incorrect anchors ... that does not appear to be the case but it certainly made the heading texts inconsistent. But heading attributes of course would actually result in wrong anchors.

Anyway I added two tests :)